### PR TITLE
Update AML to enable private IP compute

### LINKED ISSUE #2780: Cannot create private AML compute
**Describe the bug**
When trying to create AML compute with no public IP fails with

```
Provisioning error
The subnet has a network security group (/subscriptions/xxx/resourceGroups/rg-xxx-ws-
98b8/providers/Microsoft.Network/networkSecurityGroups/nsg-ws) that is
missing the following rules below. Please add those rules or increase the
priority to allow traffic if the rules already exist. For network security group
requirements, please refer to https://docs.microsoft.com/azure/machine-
learning/how-to-secure-training-vnet?tabs=azure-studio#azure-machine.
learning-compute-clusterinstance-1.
f
"name": "Outbound Storage 445"
"properties": {
"protocol": "TCP",
"sourcePortRange": "*"
"destinationPortRange": "445"
"sourceAddressPrefix": "VirtualNetwork"
"destinationAddressPrefix": "Storage OR Internet OR Storage.uksouth",
"priority": "Higher than 4096",
"direction": "Outbound"
```

Looks like the compute is trying to access public storage(?), which is disabled https://github.com/microsoft/AzureTRE/blob/ae3efacebc864ae310aa7eb753d7d002be5aff49/templates/workspace_services/azureml/terraform/network.tf#L110 

because the service isn't exposed externally. I've had a look at https://learn.microsoft.com/en-us/azure/machine-learning/how-to-secure-workspace-vnet?tabs=pe%2Ccli#azure-storage-account and there is some storage inside the ws vnet that has a private endpoint, so I'm not really any the wiser. 

Any help would be much appreciated! 

**Steps to reproduce**

1. Deploy TRE v0.6.0
2. Create workspace with AML (not exposed publicly) and VMs (exposed publicly)
3. Log into VM
4. Go to the AML URL
5. Try to create some compute, ticking the "No public IP (preview)" checkbox

### PR DESCRIPTION
**Pull request by marrobi**
Fixes #2780, #2971, #1846

## What is being addressed

Update AML workspace to cover Private IP compute scenario

## How is this addressed

- Update AML TF resource
- amend networking for additional address space and update firewall/network rules
- [x] Update documentation
- [x] Update CHANGELOG.md if needed

## Follow on work
- [ ] Configure AML compute instance user resource to pull from package mirror ( https://learn.microsoft.com/en-us/azure/machine-learning/how-to-customize-compute-instance#use-the-script-in-a-resource-manager-template ? )
- #3123
- [ ] Remove Storage firewall rules when not exposed publicly.
- [ ] Workspace diagnostics complains about build compute
- #3122

### CODE REVIEW

**Comment by marrobi** [comment]
@damoodamoo are you or one of your team still intending to take this forward? It is now usable, however a couple of outstanding issues as detailed above.

@tamirkamara if you can give this a first pass (still draft) would eb appreciated.

File: templates/workspace_services/azureml/template_schema.json [hunk 1]
```diff
@@ -3,19 +3,59 @@
...
+      "default": "Azure Machine Learning empowers data scientists and developers to build, deploy, and manage high-quality models faster and with confidence.",
+      "updateable": true
+    },
+    "overview": {
+      "type": "string",
+      "title": "Workspace Service Overview",
+      "description": "Long form description of the workspace service, in markdown syntax",
+      "default": "",
+      "updateable": true
+    },
```
Comment (tamirkamara) [inline templates/workspace_services/azureml/template_schema.json:29]: Isn't that the generic text anyway? Why should we define it in a specific template definition?

File: templates/workspace_services/azureml/terraform/compute.tf [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+}
+
+resource "azurerm_key_vault_secret" "aml_password" {
+  name         = "cp-${local.short_service_id}"
+  value        = random_password.password.result
+  key_vault_id = data.azurerm_key_vault.ws.id
+}
+
+
+resource "azapi_resource" "compute_cluster" {
```
Comment (tamirkamara) [inline templates/workspace_services/azureml/terraform/compute.tf:21]: What doesn't the native azurerm resource support that made you choose to use the azapi approach? Maybe worth documenting?

File: templates/workspace_services/azureml/terraform/outputs.tf [hunk 3]
```diff
@@ -28,6 +32,26 @@ data "azurerm_network_service_tags" "storage_tag" {
   location_filter = azurerm_storage_account.aml.location
 }
 
+data "azurerm_network_service_tags" "mcr_tag" {
```
Comment (tamirkamara) [inline templates/workspace_services/azureml/terraform/outputs.tf:35]: what is special about those kind of tags that we have to do this and not just use a name like we do elsewhere (like AzureActiveDirectory)?

Comment (marrobi) [reply templates/workspace_services/azureml/terraform/outputs.tf:35]: They are region tags to restrict to the region. Like `MicrosoftContainerRegistry.WestEurope`, this is dependant on the region being deployed to.

File: Makefile [hunk 4]
```diff
@@ -134,6 +134,10 @@ terraform-deploy:
 	$(call target_title, "Deploying ${DIR} with Terraform") \
 	&& . ${MAKEFILE_DIR}/devops/scripts/check_dependencies.sh env \
 	&& . ${MAKEFILE_DIR}/devops/scripts/load_and_validate_env.sh \
+	&& . ${MAKEFILE_DIR}/devops/scripts/load_env.sh ${DIR}/.env \
+	&& . ${MAKEFILE_DIR}/devops/scripts/load_env.sh ${MAKEFILE_DIR}/devops/.env \
+	&& . ${MAKEFILE_DIR}/devops/scripts/load_env.sh ${MAKEFILE_DIR}/templates/core/.env \
+	&& . ${MAKEFILE_DIR}/devops/scripts/load_env.sh ${MAKEFILE_DIR}/devops/auth.env \
```
[the remark this answers is not in the document]
Comment (marrobi) [reply Makefile:140]: Taken out, but there is an issue with some of these commands.

File: templates/workspace_services/azureml/template_schema.json (hunk 1 above)
Comment (marrobi) [reply templates/workspace_services/azureml/template_schema.json]: I've added a new default.

File: templates/workspace_services/azureml/terraform/compute.tf (hunk 2 above)
Comment (marrobi) [reply templates/workspace_services/azureml/terraform/compute.tf:21]: Added a comment.

**Review by tamirkamara — APPROVED** [review]

**Comment by marrobi** [comment]
/test-force-approve 

Passed: https://github.com/microsoft/AzureTRE/actions/runs/4005077826

**Pull request merged**